pyproject.toml

```diff
@@ -17,7 +17,7 @@ requires-python = ">=3.10"
 [project.scripts]
 starcraft-hello = "starcraft:hello"
 
-[project.optional-dependencies]
+[dependency-groups]
```

Comment:
I guess we will need to adapt https://github.com/canonical/starflow/blob/main/.github/workflows/scan-python.yaml#L32 too and replace with `--all-groups`? If we do that we will need to handle a transition or version the workflow, no?

Reply:
In that particular case we still want to use `--all-extras` because we want the optional extras to be included. We don't care as much about `--all-groups` for security scans because while we should update development/documentation dependencies, I don't think it's a release blocker.

Reply:
Good point. I think I agree with you.